processor/processor.go

```diff
@@ -1350,18 +1367,57 @@ func (proc *Handle) processJobsForDest(partition string, subJobs subJob, parsedE
 					source,
 				)
 
+				payloadFunc := func() json.RawMessage {
```

Comment:
first call assumes `nil` as the default and the second one assumes `{}`
```suggestion
				payloadFunc := func(default json.RawMessage) json.RawMessage {
```